segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeepertier2/StorageLedger.java

```diff
@@ -0,0 +1,183 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage.impl.bookkeepertier2;
+
+import io.pravega.common.util.ImmutableDate;
+import io.pravega.segmentstore.contracts.BadOffsetException;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import lombok.Getter;
+import org.apache.curator.framework.api.transaction.CuratorOp;
+
+/*
+* Storage Ledger represent a single segment. A segment is rolled over whenever a ownership change is observed.
+*
+**/
+public class StorageLedger {
+
+    private final StorageLedgerManager manager;
+
+    @Getter
+    private final String name;
+
+    @Getter
+    private final boolean readOnlyHandle;
+
+    @Getter
+    private int updateVersion;
+
+    @Getter
+    private long containerEpoc;
+
+    private ConcurrentHashMap<Integer, LedgerData> dataMap;
+
+    @Getter
+    private boolean sealed;
+
+    @Getter
+    private int length;
+
+    @Getter
+    private ImmutableDate lastModified;
+
+
+
+    public StorageLedger(StorageLedgerManager storageLedgerManager, String streamSegmentName, int updateVersion, long containerEpoc, boolean readOnly) {
+        manager = storageLedgerManager;
```

Comment:
1. Check for nulls and do any other meaningful parameter validation (empty strings, bad epoch, bad update version).
2. Be consistent. Either use "this." or don't, but don't mix them.

Reply:
Using this.